Revamp repo using the updated Typescript config

lewisdaly opened the pull request:
After creating the new 'official' typescript template project, I went back and applied it to this project.

Commit:
Revamp repo using the update Typescript config

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/mojaloop/central-services-health?pullRequest=2) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/mojaloop/central-services-health?pullRequest=2) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/mojaloop/central-services-health?pullRequest=2) it.</sub>

rmothilal commented on package.json:
```diff
@@ -50,20 +50,26 @@
     "pre-commit": "1.2.2",
+    "prettier": "1.18.2",
     "sinon": "7.3.2",
     "standard": "12.0.1",
```
please use latest standard library, would this be compatible with typescript?

lewisdaly replied:
Ah I probably should remove standard, as standard and typescript don't play nicely together.

lewisdaly replied:
Oh wait, I missed this last time I looked into it: https://standardjs.com/#typescript

That said, I think prettier has better integrations and seems to have more mature typescript support (especially out of the box) than standard.

rmothilal commented on package.json:
```diff
@@ -7,19 +7,19 @@
   "scripts": {
```
please update version number in package.json

rmothilal reviewed (changes requested):
please see comments

Commit:
Update package to 7.2.0, run `npm audit fix`

rmothilal commented on src/HealthCheckServer.ts:
```diff
@@ -64,7 +63,7 @@ const defaultHealthHandler = (healthCheck: any): Lifecycle.Method => {
-const failAction = async (_request: Hapi.Request, _handler: Hapi.ResponseToolkit, err?: Error) => {
+const failAction = async (_request: Hapi.Request, _handler: Hapi.ResponseToolkit, err?: Error): Promise<void> => {
   if (!err) {
     throw Boom.boomify(new Error(`Unknown Server Error`))
```
Why are you using boom errors here instead of the error handling framework

lewisdaly replied:
Will fix this.

rmothilal commented on src/HealthCheckServer.ts:
```diff
@@ -22,9 +22,8 @@
-import Boom from '@hapi/boom';
-
+import Hapi, { Lifecycle, ResponseObject } from '@hapi/hapi'
+import Boom from '@hapi/boom'
```
I don’t think boom is needed in any of our api’s anymore as we have our own error handling framework

rmothilal reviewed (changes requested):
Please see comments

Commit:
Remove Boom dependency in favour of central-services-error-handling library

Commit:
Bump package version to 7.3.0

Add audit-dependencies step and run `audit:resolve`

working on CircleCI config

working on CircleCI config

working on CircleCI config

working on CircleCI config

Commit:
working on CircleCI config

Commit:
Add typescript build config to circleci

Commit:
Add a files entry to package.json to ensure the dist folder is included in npm pack

Commit:
Merge branch 'master' of github.com:mojaloop/central-services-health into feature/update-ts-config

lewisdaly commented:
@rmothilal thanks for your feedback. I copied the `central-services-error-handling` logic from `central-ledger`, so I'm pretty sure I got it right, but please double check.

I also added a few other missing things:
- fixed missing `tsc build` in the CircleCI config; after this is merged, I'll do a snapshot release just to double check that the `npm release` works correctly
- added the `license-scanner` and `audit-licenses` steps that have been added elsewhere since this repo was created.

Commit:
Add --silent flag to npm run audit:check to ensure we print valid json

Commit:
remove unneeded caret from package.json

rmothilal commented on test/unit/HealthCheckServer.test.ts:
```diff
@@ -179,17 +172,23 @@ Test('HealthCheckServer test', function (healthCheckServerTest: any) {
-      const expected = 500
-      const request: any = null;
-      const handler: any = null;
+      const expected = '2001'
```
I would suggest using the error enums from the error library instead of hardcoding the error codes

rmothilal commented on src/HealthCheckServer.ts:
```diff
@@ -64,11 +62,8 @@ const defaultHealthHandler = (healthCheck: any): Lifecycle.Method => {
-  }
-  throw Boom.boomify(err)
+const failAction = async (_request: Hapi.Request, _handler: Hapi.ResponseToolkit, err?: Error): Promise<void> => {
+  throw ErrorHandler.Factory.reformatFSPIOPError(err)
```
If the failAction has to catch joi validation requests from http server then I could suggest using the createFSPIOPErrorFromJoi function

rmothilal reviewed (changes requested):
Please see comments

Commit:
Replace hardcoded values with enums

rmothilal reviewed (approved):
Happy
